src/krux/encryption.py

```diff
@@ -0,0 +1,282 @@
+# The MIT License (MIT)
+
+# Copyright (c) 2021-2023 Krux contributors
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+try:
+    import ujson as json
+except ImportError:
+    import json
+import hashlib
+import ucryptolib
+from .baseconv import base_encode, base_decode
+from .sd_card import SDHandler
+from .krux_settings import Settings, PBKDF2_HMAC_ECB, PBKDF2_HMAC_CBC, AES_BLOCK_SIZE
+from embit.wordlists.bip39 import WORDLIST
+
+
+MNEMONICS_FILE = "seeds.json"
+
+VERSION_MODE = {
+    "AES-ECB": ucryptolib.MODE_ECB,
+    "AES-CBC": ucryptolib.MODE_CBC,
+    PBKDF2_HMAC_ECB: ucryptolib.MODE_ECB,
+    PBKDF2_HMAC_CBC: ucryptolib.MODE_CBC,
+}
+
+VERSION_NUMBER = {
+    "AES-ECB": PBKDF2_HMAC_ECB,
+    "AES-CBC": PBKDF2_HMAC_CBC,
+}
+
+
+class AESCipher:
```

Comment:
This makes me nervous since I'm not an expert and can't really audit this. Also because we're relying on the ucryptolib which I believe was implemented by MaixPy employees...

Encryption is easy to get wrong if you don't know what you're doing. How confident are you in this implementation? Did another set of eyes look this over?

Reply:
I'm not an expert either, but Maixpy implementation seems reliable and following AES standards.
It's also working on Android App, which uses [Pycryptodome](https://pypi.org/project/pycryptodome/) and encryption from both are interoperable.
I posted an encrypted mnemonic QR code online with some sats to check if someone could hack it, no one did. A month ago a Brazilian youtube channel of education Bitcoin content with thousands of followers did the same challenge using Krux, no one hacked it too. Code and basic information were also provided.
The encryption methods are known and widely used, but off course it would be excellent to have more scrutiny on how it was implemented.